piwigo-random-backend.php

```diff
@@ -102,15 +122,20 @@ function check_param($param, $values) {
   foreach ($thumbc["result"]["images"] as $image)
   {
     $image_url = (string)$image['derivatives'][$size]['url'];
-    # Piwigo's WS returns two URLs for the image. $image['page_url'] is the URL
-    # of the image, without any category consideration, and $image['categories']
-    # is the list of categories, each of which contains the URL of the image
-    # viewed as part of this category (e.g. next/previous buttons navigate
-    # within this category). In practice, the list of categories seems to be a
-    # singleton whenever we set $cat_id, but let's have fun and pick it randomly
-    # anyway.
-    $cats = $image['categories'];
-    $page_url = (string)$cats[random_int(0, count($cats) - 1)]['page_url'];
+    if ($tag_name) {
+      $page_url = (string)$image['page_url'];
+    }
+    else {
+      # Piwigo's WS returns two URLs for the image. $image['page_url'] is the URL
```

Comment:
Just being curious: doesn't this statement also apply when searching by tag?

Perhaps a better test would be to check for the presence of the `categories` item in the result instead of testing `if ($tag_name) {` above.

Reply:
pwg.tags.getImages does not deliver categories. It's straight forward to test for $tag_name for this. Of course  $image['categories'] could be tested instead, which might be even better also for the already existing code because if it's not available or categories would be 0 the code would fail with $cats[random_int(0,-1)].
I'll change.